### DISCUSSION TITLE
Save External queries

### DISCUSSION THREAD

**Discussion by mattiason** [Budibase/budibase#5885] [Ideas]
Would it be possible to store the latest query from e.g a rest API until a new request is made? The main reason is that sometimes you would like to use some data e.g a list in a drop down and it becomes relatively slow sometimes when all the time needing to call the api. Therefore it would be nice to be able to store the last query inside the app without the need of adding it to a table.

**Comment by melohagan** [comment]
Great idea!

Somewhat related, I think being able to use <a href="https://docs.budibase.com/docs/app-state">App State</a> as a data provider would be able to accomplish this, and be a potential two for one: https://github.com/Budibase/budibase/issues/5658

Otherwise, some sort of option for REST queries to cache unless the bindings change?

**Reply by mattiason** [reply]
I think it should be possible e.g if you want to make a query once an hour, or once a day or week etc it should be able to store the resulting until the next query is made. Let's call it a collection. 

I am not so sure if the app state is the same as that seems to be more user-related where collections are more backend related ?  I maybe misunderstand something?

Mattias

**Reply by melohagan** [reply]
My thinking was that you could execute your query, and then save the result to app state, then use that app state as a data source so that a query doesn't need to be made every-time - basically using app state as a front-end cache (which can be persisted).

That may not be the best way to go about it though, and as you say an actual backend cache may be more appropriate.

**Reply by mattiason** [reply]
@melohagan  from my perspective I think it would be better as separate functions. Still, both are very important.
